internal/k8s/reconciler/route.go

```diff
@@ -371,8 +373,10 @@ func (r *K8sRoute) Validate(ctx context.Context) error {
 				if err != nil {
 					return err
 				} else if !allowed {
-					// FUTURE Set appropriate status and early out
-					r.logger.Warn("Cross-namespace routing not allowed, will enforce in future", "refName", ref.Name, "refNamespace", ref.Namespace)
+					msg := fmt.Sprintf("Cross-namespace routing not allowed without matching ReferencePolicy for Service %q", route.GetNamespace())
```

Comment:
this should say:

```suggestion
					msg := fmt.Sprintf("Cross-namespace routing not allowed without matching ReferencePolicy for Service %q", getServiceID(ref, route.GetNamespace()))
```

as below.